### ISSUE TITLE
[CT-2532] [Feature] Validate selector nodes like any other selection

### ISSUE THREAD

**Issue by mroy-seedbox** [dbt-labs/dbt-core#7513]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Let's say I have a simple selector like the one below, it will never produce a warning or error (unlike selecting the same models directly on the CLI):

```yaml
  - name: test
    definition:
      union:
        - 'model_that_exists'
        - 'model_that_does_not_exist'
```

Selecting with `-s` on the CLI:
```
> dbt --warn-error compile -s model_that_exists model_that_does_not_exist
Encountered an error:
Compilation Error
 The selection criterion 'model_that_does_not_exist' does not match any nodes
```

But when using the selector `test` (`dbt --warn-error compile --selector test`), there is no error.

I would expect the selector to produce the same error as the `-s` selection.

### Describe alternatives you've considered

Can't find or think of any. 🤷‍♂️ 

### Who will this benefit?

Everyone using selectors, to make sure their selectors are valid. Even a simple typo in a selector definition could mean one or more models are missed, but no one would ever know unless they pay close attention.

Basically, there seems to be no way to ensure that selectors are actually valid, other than through tedious manual validation. But this is not a viable way to ensure that selectors remain valid _over time_ (as models are added & removed).

### Are you interested in contributing this feature?

No

**Comment by dbeatty10** [comment]
Thanks for reaching out @mroy-seedbox !

This feature request makes sense to me, and I was able to observe the same behavior you described.

I'm guessing this would take non-trivial discovery work in order to implement and that we would not be able to prioritize it anytime soon.

### Implementation hints
Here's portions of the code that I think would be relevant:
By default [`expect_exists=False`](https://github.com/dbt-labs/dbt-core/blob/810ef7f5560b4bbe8a3afb0dc182a4380d3a3dcd/core/dbt/graph/selector_spec.py#L177) for `BaseSelectionGroup`. But when it is `True`, then I think your requested behavior would be surfaced by the following:

https://github.com/dbt-labs/dbt-core/blob/810ef7f5560b4bbe8a3afb0dc182a4380d3a3dcd/core/dbt/graph/selector.py#L144-L145

I believe the other most relevant pieces of code:
- https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/config/selectors.py
- https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/graph/cli.py

**Comment by mroy-seedbox** [comment]
As a workaround, we are now using this shell/bash command to validate the models listed in `selectors.yml`: `dbt --warn-error compile -s $(cat selectors.yml | grep -E "^[ ]*- '[^\.]*.[^\.]*.[^\.]*'$" | sed -E "s/^[ ]*- '//g" | sed -E "s/'$//g" | tr "\n" " ")`

- `$(...)` executes the command within
- `cat selectors.yml` reads the file
- `grep -E "^[ ]*- '[^\.]*.[^\.]*.[^\.]*'$"` looks for lines like `  - 'database.schema.model'` (which is how we list our models)
- `sed -E "s/^[ ]*- '//g"` removed the leading `  - '`
- `sed -E "s/'$//g"` removes the trailing `'`
- `tr "\n" " "` replaces all the newlines with spaces, so that all the models end up on the same line

End result: `dbt --warn-error compile -s model1 model2 model3 ...`

We've put that command inside of a GitHub action, so that the validation is executed on every pull request/commit. 👌 

It detects typos, missing/deleted models, etc. 👍

**Comment by lcongdanh** [comment]
I'm a new contributor but I've been learning how dbt-core works for around 5 weeks. Can I work on this?
